components/stacks-network/src/chains_coordinator.rs

```
@@ -646,6 +745,97 @@ fn should_publish_stacking_orders(
     true
 }
 
+pub async fn create_global_snapshot(
+    devnet_event_observer_config: &DevnetEventObserverConfig,
+    devnet_event_tx: &Sender<DevnetEvent>,
+    mining_command_tx: Sender<BitcoinMiningCommand>,
+) {
+    let devnet_config = &devnet_event_observer_config.devnet_config;
+    let global_snapshot_dir = get_global_snapshot_dir();
+    let project_snapshot_dir = get_project_snapshot_dir(devnet_config);
+
+    // Project snapshot marker
+    let project_marker = project_snapshot_dir.join("epoch_3_ready");
+    if !project_marker.exists() {
+        match std::fs::File::create(&project_marker) {
+            Ok(_) => {
+                let _ = devnet_event_tx.send(DevnetEvent::success(
+                                            "Project snapshot data prepared up to epoch 3.0. Future project starts will be faster.".to_string(),
+                                        ));
+            }
+            Err(e) => {
+                let _ = devnet_event_tx.send(DevnetEvent::warning(format!(
+                    "Failed to create project snapshot marker file: {}",
+                    e
+                )));
+            }
+        }
+    }
+
+    let global_marker = global_snapshot_dir.join("epoch_3_ready");
+    if !global_marker.exists() {
+        // Copy project snapshot to global snapshot as a template
+        if project_snapshot_dir != global_snapshot_dir {
+            // Copy bitcoin data
+            let project_bitcoin_snapshot = project_snapshot_dir.join("bitcoin");
+            let global_bitcoin_snapshot = global_snapshot_dir.join("bitcoin");
+            if project_bitcoin_snapshot.exists() {
+                let _ = copy_directory(&project_bitcoin_snapshot, &global_bitcoin_snapshot, None);
```

Probably shouldn't ignore errors here, this could easily fail due to a permissions error, you should at least issue a warning if that happens